pyEpiabm/pyEpiabm/tests/test_output/test_abstract_reporter.py

```diff
@@ -0,0 +1,47 @@
+import unittest
+from unittest import mock
+
+import pyEpiabm as pe
+
+
+class TestAbstractReporter(unittest.TestCase):
+    """Test the 'AbstractReporter' class.
+    """
+
+    @mock.patch("os.path.exists", return_value=False)
+    @mock.patch("os.makedirs")
+    def test_construct(self, mock_makedir, mock_pathexists):
+        reporter = pe.output.AbstractReporter("test_folder", False)
+        self.assertEqual(reporter.folder, "test_folder")
+        mock_pathexists.assert_called_once_with("test_folder")
+        mock_makedir.assert_called_once_with("test_folder")
+
+    @mock.patch("os.path.exists", return_value=True)
+    @mock.patch("os.scandir", return_value=[mock.MagicMock(),
+                                            mock.MagicMock()])
+    @mock.patch("os.remove")
+    def test_construct_clearance(self, mock_remove, mock_scandir,
+                                 mock_pathexists):
+        pe.output.AbstractReporter("test_folder", True)
+        mock_scandir.assert_called_once_with("test_folder")
+        self.assertEqual(mock_remove.call_count, 2)  # For two elements in list
+
+    @mock.patch("os.path.exists")
+    @mock.patch("os.scandir", side_effect=IsADirectoryError)
+    @mock.patch("os.remove")
```

Comment:
```suggestion
    @mock.patch("os.scandir", return_value=[mock.MagicMock()])
    @mock.patch("os.remove", side_effect=IsADirectoryError)
```
It's more realistic for the os.remove to raise the directory error